components/raftstore/src/store/peer.rs

```diff
@@ -2644,6 +2645,34 @@ where
         }
     }
 
+    pub fn handle_gen_snap_task<T: Transport>(&mut self, ctx: &mut PollContext<EK, ER, T>) {
+        if let Some(mut gen_task) = self.mut_store().take_gen_snap_task() {
+            // If the snapshot gen precheck feature is enabled, the leader needs
+            // to complete a precheck with the target follower before the
+            // snapshot generation.
+            if ctx.feature_gate.can_enable(SNAPSHOT_GEN_PRECHECK_FEATURE)
+                && !gen_task.precheck_succeeded
+            {
+                // Precheck requests are throttled. If the precheck fails
+                // because the receiver is busy, it's not necessary to retry the
+                // request too aggressively.
+                if gen_task.precheck_iter_cnt == 0 {
+                    let to_peer = gen_task.to_peer.clone();
+                    self.send_snap_gen_precheck_request(ctx, &to_peer);
+                }
+                gen_task.precheck_iter_cnt = (gen_task.precheck_iter_cnt + 1) % 20;
```

Comment:
make `20` as a const

Reply:
I struggled to find a name for that constant maybe because it's not a good parameter. I ended up defining a time interval between snapshot requests. hopefully that'll make more sense.

Reply:
Defined a constant in the latest version